### ISSUE TITLE
Bert bi encoder interactive script

### ISSUE THREAD

**Issue by SirChoeg** [facebookresearch/ParlAI#1718]
As suggested in #1681.

I tried the interactive script on the BERT bi-encoder ranker with the following command:

`python examples/interactive.py -mf models/bert_bi_daily_dialog_bs32_nostart -cands fixed -ecands fixed -fixedCands options.txt`

I used a fixed cands file as you suggested (options.txt):
```
I'm fine.
Sure. I can cook.
Don't make it harder than it needs to be.
It is stable enough.
```
But after entering a message (like "How are you?")
I get an error like this:

![image](https://user-images.githubusercontent.com/11367378/57626436-2eaacb00-7596-11e9-9725-49c73f5a8e79.png)

Is my model messed up or did I miss something else?
My best guess is that I did not format my fixed candidates file appropriately.

Thanks for your quick answer on my last issue.

**Comment by stephenroller** [comment]
I think you're giving the wrong argument. I think it's `--fixed-candidates-path filename.txt`, not `-fixedCands options.txt`. TBH I'm surprised that didn't give you an error message about not finding an argument.

https://github.com/facebookresearch/ParlAI/blob/716c253cca4779780eb5e7f76bf86210d613b957/parlai/core/torch_ranker_agent.py#L35-L38

**Comment by SirChoeg** [comment]
You were right. Thank you. Although my agent always chooses the first candidate no matter what.
![image](https://user-images.githubusercontent.com/11367378/57982173-26dda180-7a42-11e9-9a3e-eaeb877ff234.png)

**Comment by stephenroller** [comment]
8 candidates is pretty low. What are the other 7?

How do your train and validation losses look? Have you tried display_model.py with the same task it was trained on.

Usually when we see “always pick first” behavior, it’s because the candidates are all tied, or a because NaN managed to slip in somewhere and destroy the scoring.

**Comment by SirChoeg** [comment]
Candidates:
![image](https://user-images.githubusercontent.com/11367378/57983191-416a4780-7a4f-11e9-83de-e0100402802b.png)

Loss:
![image](https://user-images.githubusercontent.com/11367378/57983213-88583d00-7a4f-11e9-8bdb-7abe78e6287e.png)
Could it be ovefitting?

Display Model:
![image](https://user-images.githubusercontent.com/11367378/57983200-61017000-7a4f-11e9-984a-7dac8dc98d4e.png)

![image](https://user-images.githubusercontent.com/11367378/57983228-b76eae80-7a4f-11e9-9540-960cbf651a5d.png)
Why is there only a singe text_candidate?

**Comment by stephenroller** [comment]
It looks like around 3M epochs there was a NaN error and all hell broke loose. Can you share your train invocation?

That particular teacher doesn’t have candidates. We usually just use `-ecands batch` for those situations.

**Comment by stephenroller** [comment]
But it looks like the model isn’t learning anything at all.

**Comment by SirChoeg** [comment]
![image](https://user-images.githubusercontent.com/11367378/57984891-8ba7f480-7a60-11e9-80d8-90900e0c3288.png)

**Comment by stephenroller** [comment]
Hmm, can you put the full training log into a gist?

**Comment by stephenroller** [comment]
-histsz 1 is pretty brutal, I would remove that option but it doesn’t explain your problem here. -clip 5 is also probably better with the default but I haven’t trained a bert model so maybe that’s normal.


Try using the hyperparameters discussed in #1713. Make sure you use a new model filename when you retrain.

**Comment by stephenroller** [comment]
Closing for inactivity, but please reopen if you have more questions.
